pkg/storage/rulestore/bucket_client.go

```diff
@@ -0,0 +1,311 @@
+package rulestore
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/gogo/protobuf/proto"
+	"github.com/thanos-io/thanos/pkg/objstore"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/cortexproject/cortex/pkg/ruler/rules"
+	"github.com/cortexproject/cortex/pkg/tenant"
+)
+
+const (
+	delim      = "/"
+	rulePrefix = "rules" + delim
+)
+
+type BucketClient struct {
```

Comment:
This is not a bucket client but an implementation of the rule store. I think we should name it accordingly.

Reply:
How does `BucketRuleStore` work?